src/main.ts

```diff
@@ -2,28 +2,106 @@ import * as core from '@actions/core'
 import * as io from '@actions/io'
 import * as exec from '@actions/exec'
 import * as utils from './utils'
-import {Inputs, createPullRequest} from './github-helper'
+import {
+  Inputs,
+  createPullRequest,
+  buildBranchesFromLabels
+} from './github-helper'
 
 const CHERRYPICK_EMPTY =
   'The previous cherry-pick is now empty, possibly due to conflict resolution.'
 
+export type ExecutionStatus = {
+  failed: boolean
+  branch: string
+  inputs?: Inputs
+  msg?: string | undefined | void
+}
+
+export type Statuses = {
+  completedCherryPicks: Array<ExecutionStatus>
+  cherryPickErrors: Array<ExecutionStatus>
+}
+
 export async function run(): Promise<void> {
-  try {
-    const inputs: Inputs = {
-      token: core.getInput('token'),
-      committer: core.getInput('committer'),
-      author: core.getInput('author'),
-      branch: core.getInput('branch'),
-      labels: utils.getInputAsArray('labels'),
-      assignees: utils.getInputAsArray('assignees'),
-      reviewers: utils.getInputAsArray('reviewers'),
-      teamReviewers: utils.getInputAsArray('teamReviewers')
+  const inputs: Inputs = {
+    token: core.getInput('token'),
+    committer: core.getInput('committer'),
+    author: core.getInput('author'),
+    branch: core.getInput('branch'),
+    labels: utils.getInputAsArray('labels'),
+    assignees: utils.getInputAsArray('assignees'),
+    reviewers: utils.getInputAsArray('reviewers'),
+    teamReviewers: utils.getInputAsArray('teamReviewers'),
+    allowUserToSpecifyBranchViaLabel:
+      core.getInput('allowUserToSpecifyBranchViaLabel') || '',
+    labelPatternRequirement: core.getInput('labelPatternRequirement'),
+    userBranchPrefix: core.getInput('userBranchPrefix') || ''
+  }
+
+  const branchesToCherryPick =
+    inputs.allowUserToSpecifyBranchViaLabel === 'true'
+      ? buildBranchesFromLabels(inputs)
+      : [inputs.branch]
+  core.info(`branches to cherry pick ${JSON.stringify(branchesToCherryPick)}`)
+
+  if (!branchesToCherryPick) {
+    core.info(`No branches to cherry pick`)
+    return
+  }
+
+  const executions: Array<ExecutionStatus> = []
+
+  core.info(`Executing ${executions.length} cherry picks...`)
+
+  for (const branch of branchesToCherryPick) {
+    try {
+      executions.push(await cherryPickExecution(inputs, branch))
+    } catch (e: any) {
+      executions.push(e)
     }
```

Comment:
Executing the workflows in parallel seems to lead to weird side effects (likely because of how `gitExecution` works). Synchronous execution doesn't seem to hurt performance all that much.